solutions/dmoj-Phidias.mdx

```diff
@@ -0,0 +1,75 @@
+---
+id: dmoj-phidias
+source: DMOJ
+title: Phidas
+author: Jesse Choe
+---
+
+Let $dp[i][j]$ be the minimum residual area with a rectangular slab size of $i X j$. Initialize $dp[i][j]$ as $i * j$ and set $dp[x_i][y_i]$ as 0, where $x_i, y_i$ are the width and heights of the desired plate sizes, respectively. Check every sub-rectangle with either $i$ or $j$ fixed (but not both) and store the residual area of the optimal sub-rectangle.
+
+**Time Complexity:** $\mathcal{O}(W * H * (W + H))$.
+
+```cpp
+#include <bits/stdc++.h>
+ 
+using namespace std;
+using ll = long long;
+using vi = vector<int>;
+using pi = pair<int, int>;
+ 
+#define f first
+#define s second
+
+/*
+ * Let dp[i][j] be the minimum residual area with a 
+ * rectangular slab size of i X j
+ */
+ 
+const int MAXN = 601;
+int dp[MAXN][MAXN];
+
+int main(){
+	// The width and height of the original slab.
+	int w, h; cin >> w >> h;
+	int n; cin >> n; vector<pi> plates(n);
+	for(int i = 0; i < n; i++){
```

Comment:
1. statements each on one line
2. spaces for the control flow statements

Reply:
wdym

Reply:
1. instead of `int n; cin >> n; vector<pi> plates(n);` do something like
```cpp
int n;
cin >> n;
vector<pi> planes(n);
```
2. format your control statements like this:
```cpp
for (int i = 0; i < n; i++) {
}
```